src/Storages/StorageMemory.cpp

```diff
@@ -320,21 +319,25 @@ namespace
             IndexForNativeFormat index;
             {
                 auto data_file_path = temp_dir / fs::path{file_paths[data_bin_pos]}.filename();
-                auto data_out_compressed = temp_disk->writeFile(data_file_path);
-                CompressedWriteBuffer data_out{*data_out_compressed, CompressionCodecFactory::instance().getDefaultCodec(), max_compress_block_size};
-                NativeWriter block_out{data_out, 0, metadata_snapshot->getSampleBlock(), false, &index};
-                for (const auto & block : *blocks)
-                    block_out.write(block);
-                backup_entries[data_bin_pos] = {file_paths[data_bin_pos], std::make_shared<BackupEntryFromImmutableFile>(temp_disk, data_file_path, read_settings)};
+                {
+                    auto data_out_compressed = temp_disk->writeFile(data_file_path);
+                    CompressedWriteBuffer data_out{*data_out_compressed, CompressionCodecFactory::instance().getDefaultCodec(), max_compress_block_size};
+                    NativeWriter block_out{data_out, 0, metadata_snapshot->getSampleBlock(), false, &index};
+                    for (const auto & block : *blocks)
+                        block_out.write(block);
+                }
```

Comment:
Interesting why do we need additional scope here? Do we rely on that after this scope some file descriptors will be closed? If so, comment would be good.

Reply:
We need `~data_out` to call `data_out.finalize()`.